/* Copyright 2019 Alexander Belinsky. All Rights Reserved.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at
    http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
==============================================================================*/

syntax = "proto3";

package energyplatform;
import "energyplatform/core/action.proto";
import "energyplatform/core/observation.proto";


////////////////////////////////////////////////////////////////////////////////
//
// PredctorService defines a service that predicts actions based on specified observations.
//
////////////////////////////////////////////////////////////////////////////////

message PredictRequest {
    // current observation
    Observation observation = 1;
}

message PredictReponse {
    // recommended action
    Action action = 1;
}

// Main class for predictor services through platform.
service PredictorService {
    // Predicts actions
    rpc Predict(PredictRequest) returns (PredictReponse);
}
